test/core/storages/dataset.test.ts

```diff
@@ -296,10 +296,9 @@ describe('dataset', () => {
                     item.index = index;
                     item.bar = 'xxx';
 
-                    // @ts-expect-error FIXME the inference is broken for `reduce()` method
                     return memo.concat(item);
                 },
-                [],
+                new Array(),
```

Comment:
this is actually fixing something? how is `[]` different from `new Array()`?

Reply:
Sorry, this should have gone here: https://github.com/apify/crawlee/pull/2774#issuecomment-2558173525